### DISCUSSION TITLE
Is public const bad?

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#12] [General]
# Is public const bad?

Is declaring a number or string as public const considered bad practice? Let's have a look what a const variable means in the first place.
Let's find out and also check what are the alternatives.

https://steven-giesel.com/blogPost/bfb3e48e-c18b-40f1-97f8-b4c8d97a0f95

**Comment by navidyazdi** [comment]
In my opinion, one of the best usage of constants is to use them as enum in models. Example: We have a class called Post and it has a property called PostType. The PostType holds a few values such as News, Article, PremiumPost and etc.
I create an enum with PostType name and add the values, then create the PostType property of the Post class with the type of PostType.
Do you agree that this is one of the most important use cases of constants?

**Reply by linkdotnet** [reply]
Yes that makes sense. If you have some constants which are related to another `enum` is a good way of showing that. 100% agreed.
